provider/cmd/pulumi-resource-random/schema.json

```diff
@@ -227,151 +183,162 @@
             "properties": {
+                "bcryptHash": {
+                    "type": "string",
+                    "description": "A bcrypt hash of the generated random string.\n",
+                    "secret": true
+                },
                 "keepers": {
                     "type": "object",
                     "additionalProperties": {
-                        "$ref": "pulumi.json#/Any"
+                        "type": "string"
                     },
-                    "description": "Arbitrary map of values that, when changed, will trigger recreation of resource. See [the main provider\ndocumentation](../index.html) for more information.\n"
+                    "description": "Arbitrary map of values that, when changed, will trigger recreation of resource. See the main provider documentation for more information.\n"
                 },
                 "length": {
-                    "type": "integer",
-                    "description": "The length of the string desired.\n"
+                    "type": "number",
+                    "description": "The length of the string desired. The minimum value for length is 1 and, length must also be \u003e= (`min_upper` + `min_lower` + `min_numeric` + `min_special`).\n"
                 },
                 "lower": {
                     "type": "boolean",
-                    "description": "Include lowercase alphabet characters in the result.\n"
+                    "description": "Include lowercase alphabet characters in the result. Default value is `true`.\n"
```

Comment:
Hm, the use of defaults here makes checking up on default value application a blocker I think, that at least it works in basic form.